### PR TITLE
Fix UBSan on CI

### PR DESCRIPTION
**Pull request by tlively**
The UBSan builder started failing with an error about a misaligned store
in wasm-ctor-eval.cpp. The store was already done via `memcpy` to avoid
alignment issues, but apparently this is no longer enough. Use `void*`
as the destination type to further avoid giving the impression of
guaranteed alignment.

Also fix UB when executing std::abs on minimum negative integers in
literal.cpp.

### CODE REVIEW

**Comment by tlively** [comment]
Wooo it works! @kripken PTAL at the latest state.

File: src/wasm/literal.cpp [hunk 1]
```diff
@@ -978,8 +978,14 @@ Literal Literal::neg() const {
...
   switch (type.getBasic()) {
     case Type::i32:
+      if (i32 == std::numeric_limits<int32_t>::min()) {
+        return *this;
+      }
       return Literal(std::abs(i32));
     case Type::i64:
+      if (i64 == std::numeric_limits<int64_t>::min()) {
+        return *this;
+      }
```
Comment (kripken) [inline src/wasm/literal.cpp:988]: Wait, wasm doesn't have integer `min`, does it? Unless this is for SIMD somehow?

**Review by kripken — COMMENTED** [review]
Any idea why this broke, btw? Compiler update on CI maybe?

File: src/wasm/literal.cpp (hunk 1 above)
Comment (kripken) [reply src/wasm/literal.cpp:988]: Oh, it's not a wasm instruction, it's some internal computation over literals... in that case we don't have a spec to compare against. Why is returning the (negative) literal the right value for `abs` here? (I would guess maxint)

Comment (kripken) [reply src/wasm/literal.cpp:988]: Out of curiosity, I found a testcase that uses this,
```wat
 (func $optimize-boolean (type $1) (param $0 i32) (param $1 i64) (result i32)
  (select
   (i32.const 0)
   (i32.const 0)
   (i32.rem_s
    (local.get $0)
    (i32.const 0)
   )
  )
 )
```
`--optimize-instructions` ends up doing `abs` on an integer here.

Reading the pass source, I'm not sure if this is a bug or not.

If you can confirm this PR doesn't change the outcome, that sounds ok, but I guess it's UB so we aren't sure it was consistent before?

Comment (tlively) [reply src/wasm/literal.cpp:988]: There is an integer abs SIMD instruction, and we run a spec test for its behavior that shows this is WAI.

**Comment by tlively** [comment]
Yeah they updated the latest ubuntu image recently, and I think it brought in subtly different versions of various things.

File: src/wasm/literal.cpp (hunk 1 above)
Comment (kripken) [reply src/wasm/literal.cpp:988]: Ok, sgtm then. Meanwhile I verified all uses of integer `abs` in `OptimizeInstructions` are valid.

**Review by kripken — APPROVED** [review]

**Pull request merged**